src/traceViewer/web/ui/propertiesTabbedPane.ts

```diff
@@ -75,16 +80,55 @@ class SourceTab {
   label = 'Source';
 
   readonly _element: Element$;
+  private _editor: monaco.editor.IStandaloneCodeEditor;
+  private _fileName: string | undefined;
+  private _decorations: string[] = [];
 
   constructor() {
-    this._element = dom`<div></div>`;
+    this._element = dom`<vbox></vbox>`;
+    monaco.editor.setTheme('vs-dark');
+    this._editor = monaco.editor.create(this._element, {
+      value: '',
+      language: 'javascript',
+      readOnly: true
+    });
+    window.addEventListener('resize', () => this.resize());
```

Comment:
I need that in Try Playwright!

Reply:
You will actually want the entire thing hosted there because it'll give screenshots, snapshots, logs, etc.